src/pybind/mgr/cephadm/designs/host-maintenance.rst

```diff
@@ -0,0 +1,100 @@
+================
+Host Maintenance
+================
+
+All hosts that support Ceph daemons need to support maintenance activity, whether the host
+is physical or virtual(vm or cloud). This means that the management workflows should provide
+a simple and consistent way to support this operational requirement. This document defines
+the maintenance strategy that will be implemented in cephadm and mgr/cephadm.
+
+
+High Level Design
+=================
+Placing a host into maintenance, adopts the following workflow;
+
+#. confirm that the removal of the host does not impact data availability (the following
+   steps will assume it is safe to proceed)
+
+   * orch host ok-to-stop <host> would be used here
+
+#. if the host has osd daemons, apply noout to the host subtree to prevent data migration
+   from triggering during the planned maintenance slot.
+#. Stop the ceph target (all daemons stop)
+#. Disable the ceph target on that host, to prevent a reboot from automatically starting
+   ceph services again)
+
+
+Exiting Maintenance, is basically the reverse of the above sequence
+
+Admin Interaction
+=================
+The ceph orch command will be extended to support maintenance. 
+
+.. code-block::
+
+    ceph orch host enter-maintenance <host> [ --check ]
+    ceph orch host exit-maintenance <host> 
```

Comment:
similar to `kubectl cordon` ?

Reply:
Kind of, but cordon marks a node as unschedulable for new pods - what I'm advocating here is that we stop our services, place the host in maintenance and hand the host to the admin. This enables them to get the maintenance done, be that OS reboot, or component upgrade/replacement - and the ceph control plane (orchestrator) knows what's going on since the host is in a maintenance state.